src/push.rs

```diff
@@ -215,6 +223,16 @@ pub async fn build_profile(data: PushProfileData<'_>) -> Result<(), PushProfileE
         &data.deploy_data.profile.profile_settings.path
     );
 
+    let nix_version_output = Command::new("nix").arg("--version")
+        .output()
+        .await.map_err(PushProfileError::NixVersion)?;
```

Comment:
stylistically, I would put the `await` either on it's own line or behind `output()`